frontend/javascript/page_navigation.js

```diff
@@ -16,6 +16,73 @@ export function enableScrollToTop() {
   });
 }
 
+export function setupSidebarItemEventListeners() {
+  const sidebarItems = document.querySelectorAll('.sidebar-item-h2');
+
+  sidebarItems.forEach(function(item) {
+    item.addEventListener('mouseenter', function() {
+      item.style.color = '#34d399';
+    });
+    item.addEventListener('mouseleave', function() {
+      item.style.color = 'gray';
+    });
+    item.addEventListener('click', function() {
+      const currentVersion = item.textContent;
+      const headings = document.querySelectorAll('h2');
+      let targetElement = null;
+      headings.forEach(function(heading) {
+        if (heading.textContent.trim() === currentVersion.trim()) {
+          targetElement = heading;
+        }
+      });
+      if (targetElement) {
+        window.scrollTo(0, targetElement.offsetTop - 100);
+      }
+    });
+  });
+}
+
+export function setupSidebar() {
+  const h1Elements = document.querySelectorAll('h1');
+  const h2Elements = document.querySelectorAll('h2');
```

Comment:
changing this line to `document.querySelectorAll('article h2')` would mean you can get rid of the counting of categories down below. That also future-proofs your changes in case the number of h2 tags in the navigation panel changes. you can't trust that the number of elements on the page will remain fixed. and _if you have to for some reason you have to leave a comment explaining your decision_